### PR TITLE
Document Personal Access Tokens

### PR DESCRIPTION
**Pull request by adamlazik1**
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.6/Katello 4.8
* [x] Foreman 3.5/Katello 4.7 (planned Satellite 6.13)
* [x] Foreman 3.4/Katello 4.6 (EL8 only)
* [x] Foreman 3.3/Katello 4.5 on EL7 & EL8 (Satellite 6.12 on EL8 only)
* [x] Foreman 3.2/Katello 4.4 on EL7 & EL8
* [x] Foreman 3.1/Katello 4.3 on EL7 & EL8 (Satellite 6.11 EL7/8, orcharhino 6.2 on EL7/8)
* For Foreman 3.0 or older, please create a separate PR.
* We do not accept PRs for Foreman 2.4 or older.

### CODE REVIEW

File: guides/common/attributes-base.adoc [hunk 1]
```diff
@@ -76,6 +76,7 @@
 :package-remove: dnf remove
 :package-update-project: dnf update
 :package-update: dnf update
+:PAT: Personal Access Token
```
Comment (maximiliankolb) [inline guides/common/attributes-base.adoc:79]: I haven't found a case where you overwrite PAT > Do you really need this as an attribute?

File: guides/common/modules/con_using-personal-access-tokens.adoc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+[id="using-personal-access-tokens_{context}"]
+= Using {PAT}s
+
+{PAT}s allow you to authenticate API requests without using your password.
+You can set an expiration date for your {PAT}, and you can also revoke it if you decide it should expire before the expiration date.
```
Comment (maximiliankolb) [inline guides/common/modules/con_using-personal-access-tokens.adoc:5]:
```suggestion
You can set an expiration date for your {PAT} and you can revoke it if you decide it should expire before the expiration date.
```

File: guides/common/modules/proc_creating-a-personal-access-token.adoc [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+. Enter a *Name* for you {PAT}.
+. Optional: Select the *Expires* date to set an expiration date.
+If you do not set an expiration date, {PAT} will never expire unless revoked.
+. Click Submit.
+You now have the {PAT} available to you on the *{PAT}s* tab.
+. Click *Copy to clipboard* to copy your {PAT}.
++
+[IMPORTANT]
+====
+Make sure to store your {PAT} as you won't be able to access it again after you leave the page or create a new {PAT}.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_creating-a-personal-access-token.adoc:19]:
```suggestion
Ensure to store your {PAT} as you won't be able to access it again after you leave the page or create a new {PAT}.
```

File: guides/common/modules/proc_creating-a-personal-access-token.adoc [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+# curl https://<em>{foreman-example-com}</em>/api/status --user <em>USERNAME</em>:<em>PERSONAL_ACCESS_TOKEN</em>
+----
+. You should receive a response with status `200`, for example:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+{"{project-context}_version":"{ProductVersion}.0","result":"ok","status":200,"version":"3.5.1.10","api_version":2}
+----
++
+If you go back to *{PAT}s* tab, you can also see a new *Last Used* time next to your {PAT}.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_creating-a-personal-access-token.adoc:37]:
```suggestion
If you go back to *{PAT}s* tab, you can see the *Last Used* time next to your {PAT}.
```

File: guides/common/modules/proc_revoking-a-personal-access-token.adoc [hunk 5]
```diff
@@ -0,0 +1,27 @@
+[id="revoking-a-personal-access-token_{context}"]
+= Revoking a {PAT}
+
+Use this procedure to revoke a {PAT} before its expiration date.
+
+.Procedure
+. Navigate to *Administer* > *Users*.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_revoking-a-personal-access-token.adoc:7]:
```suggestion
. In the {ProjectWebUI}, navigate to *Administer* > *Users*.
```

File: guides/common/modules/proc_creating-a-personal-access-token.adoc [hunk 6]
```diff
@@ -0,0 +1,37 @@
+[id="creating-a-personal-access-token_{context}"]
+= Creating a {PAT}
+
+Use this procedure to create a {PAT}.
+
+.Procedure
+. Navigate to *Administer* > *Users*.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_creating-a-personal-access-token.adoc:7]:
```suggestion
. In the {ProjectWebUI}, navigate to *Administer* > *Users*.
```

File: guides/common/modules/proc_revoking-a-personal-access-token.adoc [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+.Verification
+. Make an API request to your {ProjectServer} and try to authenticate with the revoked {PAT}:
+// tags in combination with `none` substitution have to be used here to render colon between USERNAME and PAT as normal.
++
+[options="nowrap", subs="none,attributes"]
+----
+# curl https://<em>{foreman-example-com}</em>/api/status --user <em>USERNAME</em>:<em>PERSONAL_ACCESS_TOKEN</em>
+----
+. You should receive an error message saying:
```
Comment (maximiliankolb) [inline guides/common/modules/proc_revoking-a-personal-access-token.adoc:20]: Please reword/drop "should".

**Review by maximiliankolb — CHANGES_REQUESTED** [review]

**Comment by adamlazik1** [comment]
@maximiliankolb Thank you for the review. Applied suggestions and ready for re-review.

File: guides/common/modules/proc_creating-a-personal-access-token.adoc [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+.Procedure
+. In the {ProjectWebUI}, navigate to *Administer* > *Users*.
+. Select a user for which you want to create a {PAT}.
+. On the *{PAT}s* tab, click *Add {PAT}*.
+. Enter a *Name* for you {PAT}.
+. Optional: Select the *Expires* date to set an expiration date.
+If you do not set an expiration date, {PAT} will never expire unless revoked.
+. Click Submit.
+You now have the {PAT} available to you on the *{PAT}s* tab.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_creating-a-personal-access-token.adoc:14]:
```suggestion
```

I am unsure but lean towards dropping this line.

File: guides/common/modules/proc_creating-a-personal-access-token.adoc [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+. Click *Copy to clipboard* to copy your {PAT}.
++
+[IMPORTANT]
+====
+Ensure to store your {PAT} as you won't be able to access it again after you leave the page or create a new {PAT}.
+====
+
+.Verification
+. Make an API request to your {ProjectServer} and authenticate with your new {PAT}:
```
Comment (maximiliankolb) [inline guides/common/modules/proc_creating-a-personal-access-token.adoc:23]:
```suggestion
. Make an API request to your {ProjectServer} and authenticate with your {PAT}:
```

**Review by maximiliankolb — COMMENTED** [review]
Made some more small suggestions. I am unsure about the usage of `<em>`; can you please try `__` due to the underscores in `Personal_Access_Token`?

File: guides/common/modules/proc_creating-a-personal-access-token.adoc (hunk 8 above)
Comment (adamlazik1) [reply guides/common/modules/proc_creating-a-personal-access-token.adoc:14]: I'd prefer to keep this line to prompt the user to look for the PAT as they won't see it again.

File: guides/common/modules/proc_creating-a-personal-access-token.adoc [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+. Enter a *Name* for you {PAT}.
+. Optional: Select the *Expires* date to set an expiration date.
+If you do not set an expiration date, {PAT} will never expire unless revoked.
+. Click Submit.
+You now have the {PAT} available to you on the *{PAT}s* tab.
+. Click *Copy to clipboard* to copy your {PAT}.
++
+[IMPORTANT]
+====
+Ensure to store your {PAT} as you won't be able to access it again after you leave the page or create a new {PAT}.
```
[the remark this answers is not in the document]
Comment (adamlazik1) [reply guides/common/modules/proc_creating-a-personal-access-token.adoc:19]: I'd like to avoid using shortened forms as they sound less official (I also don't like me sounding like a bureaucrat
 here). Maybe we can create an issue to have further discussion and then add the agreed-upon rules to README?

**Comment by adamlazik1** [comment]
> I am unsure about the usage of `\<em\>`; can you please try `__` due to the underscores in `Personal_Access_Token`?

I tried that initially, but kept getting *USERNAME*:_PERSONAL_ACCESS_TOKEN_ or something similar. It took me quite a while to come up with formatting that works.

File: guides/common/modules/proc_creating-a-personal-access-token.adoc (hunk 8 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_creating-a-personal-access-token.adoc:14]: Maybe reword to "Copy your PAT to your clipboard/Note the PAT because you cannot access it after leaving this page."? Or maybe just drop the admonition and use line 19 as explanation?

File: guides/common/modules/proc_creating-a-personal-access-token.adoc (hunk 10 above)
[the remark this answers is not in the document]
Comment (adamlazik1) [reply guides/common/modules/proc_creating-a-personal-access-token.adoc:19]: That being said, I had a brain-freeze and didn't realize I was the one using shortened form. My apologies. No discussion needed.

File: guides/common/modules/proc_creating-a-personal-access-token.adoc (hunk 8 above)
Comment (adamlazik1) [reply guides/common/modules/proc_creating-a-personal-access-token.adoc:14]: How about this: I removed the last step and made it part of the admonition.

I imagine that if I was a user and there was not a big **IMPORTANT** admonition striking my eyes, I would definitely lose my first PAT expecting it to stick around forever. (I actually did expect it to stick around forever when testing the procedure.)

**Comment by adamlazik1** [comment]
@maximiliankolb is this good to move forward?

**Review by maximiliankolb — APPROVED** [review]
LGTM.

**Comment by adamlazik1** [comment]
Merged and cherry-picked:
   5c817b871..ab4a66899  3.1 -> 3.1
   8aaf269a5..1d6318794  3.2 -> 3.2
   d8717e65a..465dc2ff9  3.3 -> 3.3
   13eb367af..847f4e7f2  3.4 -> 3.4
   34d2053c9..6f25b5ba9  3.5 -> 3.5
   3b35fd39b..704a3c2c6  3.6 -> 3.6
